Sources/Account/Account+SCIOperations.swift

```diff
@@ -0,0 +1,162 @@
+            targetQueue: DispatchQueue?
+        ) {
+            self.serialQueue = DispatchQueue(
+                label: "com.mobilecoin.\(Account.self).\(Self.self))",
+                target: targetQueue)
+            self.account = account
+            self.metaFetcher = metaFetcher
+            self.txOutSelector = TxOutSelector(txOutSelectionStrategy: txOutSelectionStrategy)
+            self.signedContingentInputCreator = SignedContingentInputCreator(
+                accountKey: account.accessWithoutLocking.accountKey,
+                fogMerkleProofService: fogMerkleProofService,
+                fogResolverManager: fogResolverManager,
+                mixinSelectionStrategy: mixinSelectionStrategy,
+                rngSeed: rngSeed,
+                targetQueue: targetQueue)
+        }
+
+        func createSignedContingentInput(
+            to recipient: PublicAddress,
+            memoType: MemoType,
+            amountToSend: Amount,
+            amountToReceive: Amount,
+            completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>
+            ) -> Void
+        ) {
+            guard amountToSend.value > 0 else {
+                let errorMessage = "createSignedContingentInput failure: " +
+                    "Cannot spend 0 \(amountToSend.tokenId)"
+                logger.error(errorMessage, logFunction: false)
+                serialQueue.async {
+                    completion(.failure(.invalidInput(errorMessage)))
+                }
+                return
+            }
+
+            guard amountToReceive.value > 0 else {
+                let errorMessage = "createSignedContingentInput failure: " +
+                    "Cannot receive 0 \(amountToReceive.tokenId)"
+                logger.error(errorMessage, logFunction: false)
+                serialQueue.async {
+                    completion(.failure(.invalidInput(errorMessage)))
+                }
+                return
+            }
+
+            // get all unspent txouts (getUnspentTxOuts() ?)
+            let (unspentTxOuts, ledgerBlockCount) =
+            account.readSync {
+                ($0.unspentTxOuts(tokenId: amountToSend.tokenId), $0.knowableBlockCount)
+            }
+            logger.info(
+                "Creating signed contingent input to recipient: \(redacting: recipient), " +
+                    "amountToSpend: \(redacting: amountToSend), " +
+                    "amountToRecieve: \(redacting: amountToReceive), " +
+                    "unspentTxOutValues: \(redacting: unspentTxOuts.map { $0.value })",
+                logFunction: false)
+
+            // fee is zero here, because the fee will be covered by the consumer of the SCI
+            switch txOutSelector
+                .selectTransactionInputs(amount: amountToSend, fee: 0, fromTxOuts: unspentTxOuts)
+                .mapError({ error -> TransactionPreparationError in
+                    switch error {
```

Comment:
do you need the re-map the error here if you create a new on in the failure case ?